src/algebra/homology/exact.lean

```diff
@@ -83,6 +83,35 @@ lemma preadditive.exact_iff_homology_zero {A B C : V} (f : A ⟶ B) (g : B ⟶ C
     exact ⟨w, preadditive.epi_of_cokernel_zero ((cancel_mono i.hom).mp (by ext))⟩,
   end⟩
 
+lemma preadditive.exact_of_iso_of_exact {A₁ B₁ C₁ A₂ B₂ C₂ : V}
+  (f₁ : A₁ ⟶ B₁) (g₁ : B₁ ⟶ C₁) (f₂ : A₂ ⟶ B₂) (g₂ : B₂ ⟶ C₂)
+  (α : arrow.mk f₁ ≅ arrow.mk f₂) (β : arrow.mk g₁ ≅ arrow.mk g₂) (p : α.hom.right = β.hom.left)
+  (h : exact f₁ g₁) :
+  exact f₂ g₂ :=
+begin
+  rw preadditive.exact_iff_homology_zero at h ⊢,
+  rcases h with ⟨w₁, ⟨i⟩⟩,
+  suffices w₂ : f₂ ≫ g₂ = 0, from ⟨w₂, ⟨(homology.map_iso w₁ w₂ α β p).symm.trans i⟩⟩,
+  rw [← cancel_epi α.hom.left, ← cancel_mono β.inv.right, comp_zero, zero_comp],
+  erw [arrow.w_assoc α.hom, p, category.assoc, category.assoc, ← arrow.w β.inv],
+  slice_lhs 2 3 { rw [← comma.comp_left, β.hom_inv_id] },
+  erw [arrow.id_left, category.id_comp],
+  exact w₁
```

Comment:
In case you are interested in an `erw`-free proof: if you add the lemma
```lean
@[simp] lemma left_hom_inv_right [is_iso sq] : sq.left ≫ g.hom ≫ inv sq.right = f.hom :=
by simp only [←category.assoc, is_iso.comp_inv_eq, w]
```
after `inv_right` in `category_theory/arrow.lean`, then you can use the following proof:

```suggestion
  rw [← cancel_epi α.hom.left, ← cancel_mono β.inv.right, comp_zero, zero_comp, ← w₁],
  simp only [← arrow.mk_hom f₁, ← arrow.left_hom_inv_right α.hom,
      ← arrow.mk_hom g₁, ← arrow.left_hom_inv_right β.hom, p],
  simp only [arrow.mk_hom, is_iso.inv_hom_id_assoc, category.assoc, ← arrow.inv_right,
    is_iso.iso.inv_hom]
```
It is probably possible to golf this.